### PR TITLE
backtester: Improve test coverage

### PR DESCRIPTION
**Pull request by lrascao**
# PR Description

Slightly improves test coverage in some backtester packages.

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [X] Bug fix (non-breaking change which fixes an issue)

## How has this been tested

- [X] go test ./... -race
- [X] golangci-lint run

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: backtester/data/fake.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
+package data
```
Comment (shazbert) [inline backtester/data/fake.go:1]: I think it might be better adding all the struct and methods to a test file.

**Review by shazbert — CHANGES_REQUESTED** [review]
Thank you for adding tests and going over the backtester code base. Just one minor issue, everything else looks good from me. 👍

File: backtester/data/fake.go (hunk 1 above)
Comment (lrascao) [reply backtester/data/fake.go]: roger that, updated

**Review by shazbert — APPROVED** [review]
changes ACK! Thank you very much for your contribution!

**Review by gloriousCode — APPROVED** [review]
tACK! Thank you for your contribution Irascao! 
# 🎉

**Review by thrasher- — APPROVED** [review]
tACK, thanks for your contribution @lrascao !

**Pull request merged**